tools/SetupFlow/DevHome.SetupFlow/ViewModels/SetupFlowViewModel.cs

```diff
@@ -149,43 +146,77 @@ public void OnNavigatedTo(NavigationEventArgs args)
+            actionPair.Value(parameter);
         }
-        else if (args.Parameter is object[] configObjs && configObjs.Length == 3)
+        else
         {
-            if (configObjs[0] is string configObj && configObj.Equals(_configurationFlowNavigationParameter, StringComparison.OrdinalIgnoreCase))
+            if (args.Parameter is object[] configObjs && configObjs.Length == 3)
             {
-                // We expect that when navigating from anywhere in Dev Home to the create environment page
-                // that the arg.Parameter variable be an object array with the the first value being 'StartCreationFlow',
-                // the second value being the page name that redirection came from for telemetry purposes, and
-                // the third value being the ComputeSystemReviewItem to setup.
-                Cancel();
-                StartSetupFlow(originPage: configObjs[1] as string, item: configObjs[2] as ComputeSystemReviewItem);
+                if (configObjs[0] is string configObj && configObj.Equals(_configurationFlowNavigationParameter, StringComparison.OrdinalIgnoreCase))
+                {
+                    didNavigate = true;
+
+                    // We expect that when navigating from anywhere in Dev Home to the create environment page
+                    // that the arg.Parameter variable be an object array with the the first value being 'StartCreationFlow',
+                    // the second value being the page name that redirection came from for telemetry purposes, and
+                    // the third value being the ComputeSystemReviewItem to setup.
+                    Cancel();
+                    StartSetupFlow(originPage: configObjs[1] as string, item: configObjs[2] as ComputeSystemReviewItem);
+                }
             }
         }
+
+        if (!didNavigate)
+        {
+            _log.Warning($"Did not navigate with args {parameter}");
+        }
     }
 
     public void StartAppManagementFlow(string query = null)
     {
         Orchestrator.FlowPages = [_mainPageViewModel];
         _mainPageViewModel.StartAppManagementFlow(query);
     }
+
+    private void StartCreationFlow(string parameter)
+    {
+        // We expect that when navigating from anywhere in Dev Home to the create environment page
+        // that the arg.Parameter variable be semicolon delimited string with the first value being 'StartCreationFlow'
+        // and the second value being the page name that redirection came from for telemetry purposes.
+        var parameters = parameter.Split(';');
+        Cancel();
+
+        Orchestrator.FlowPages = [_mainPageViewModel];
+
+        // This method is only called when the user clicks a button that redirects them to 'Create Environment' flow in the setup flow.
+        _mainPageViewModel.StartCreateEnvironmentWithTelemetry(string.Empty, _creationFlowNavigationParameter, parameters[1]);
+    }
+
+    private void StartQuickTestFlow(string parameter)

```

Comment:
`Test` in the name?

Reply:
I guess I added "Test" in the name without realizing it.  Changed.